dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterSchedulerService.java

```diff
@@ -179,10 +179,6 @@ private void scheduleProcess() throws Exception {
     }
 
     private List<ProcessInstance> command2ProcessInstance(List<Command> commands) {
-        if (CollectionUtils.isEmpty(commands)) {
-            return null;
-        }
-
         List<ProcessInstance> processInstances = new ArrayList<>(commands.size());
```

Comment:
it returns NPE when commands is null

Reply:
first we know the function is for transform the command to process instance,
so we assue the command is not null, and also the we can know that the function who call the command2ProcessInstance have the predicate that the command is null or not null

![image](https://user-images.githubusercontent.com/8596901/155912368-d66b2b6e-375d-478d-b135-fe456f8e1e36.png)